Adds a section for integration in mvc and expressive applications (#183)
zendframework/zend-inputfilter: docs/book/application-integration/usage-in-a-zend-mvc-application.md

# Usage in a zend-mvc Application

> The following example shows _one_ usage of zend-inputfilter in a zend-mvc

Reviewer: ```suggestion
The following example shows _one_ potential use case of zend-inputfilter within a zend-mvc
```

based application. The example uses a module, a controller and the
zend-inputfilter plugin manager.

Before starting, make sure zend-inputfilter is installed and configured.

> ## Create Input-Filter

Reviewer: Prior to this in the document, you use "input filter"; from here forward, you use "input-filter". Update from here forward, please.

    }
}
```

### Create Factory for Controller

Fetch the `QueryInputFilter` from the input filter plugin manager in a factory,
e.g. `src/Album/Handler/ListHandlerFactory.php`:

```php
namespace Album\Controller;

use Album\InputFilter\QueryInputFilter;
use Interop\Container\ContainerInterface;
use Zend\InputFilter\InputFilterPluginManager;
use Zend\ServiceManager\Factory\FactoryInterface;

class AlbumControllerFactory implements FactoryInterface
{
    public function __invoke(
        ContainerInterface $container,
        $requestedName,
        array $options = null
    ) {
        /** @var InputFilterPluginManager $pluginManager */
        $pluginManager = $container->get(InputFilterPluginManager::class);
        $inputFilter   = $pluginManager->get(QueryInputFilter::class);

        return new AlbumController($inputFilter);
    }
}
```

> ### Instantiating the InputFilter
>
> The `InputFilterPluginManager` is used instead of directly instantiating the
> input filter to ensure we get the filter and validator plugin managers
> injected. This allows to use custom registered filters and validators.
>
> Additionally the `InputFilterPluginManager` calls the `init` method _after_

> > instantiating the input filter.

Reviewer: Copy and paste the changes from the previous iteration of this note here, to ensure they are consistently worded. Or link to the original note.
